test/ngSanitize/sanitizeSpec.js

```diff
@@ -280,10 +280,51 @@ describe('HTML', function() {
     expect(doc).toEqual('<p><img src="x"></p>');
   }));
 
+  describe('Custom white-list support', function() {
+    beforeEach(module(function($sanitizeProvider) {
+      $sanitizeProvider.addValidElements(['form']);
+
+      $sanitizeProvider.addValidElements({
+        htmlElements: ['button', 'video'],
+        htmlVoidElements: ['input'],
+        svgElements: ['font-face-uri']
+      });
+
+      $sanitizeProvider.addValidAttrs(['required']);
+    }));
+
+    it('should allow custom white-listed element', function() {
+      expectHTML('<form></form>').toEqual('<form></form>');
+    });
+
+    it('should allow custom white-listed inline element', function() {
+      expectHTML('<button></button>').toEqual('<button></button>');
+    });
+
+    it('should allow custom white-listed block element', function() {
+      expectHTML('<video></video>').toEqual('<video></video>');
+    });
+
+    it('should allow custom white-listed void element', function() {
+      expectHTML('<input/>').toEqual('<input>');
```

Comment:
There should be a test that this is also allowed as non-void element.

Reply:
I have added check 

```
      expectHTML('<input/>').not.toEqual('<input></input>');
```

Reply:
No, I meant something like `expectHTML('<input></input>')` (which is accepted atm).

Reply:
🤔  I didn't get your point.

> There should be a test that this is also allowed as non-void element.

We already have such test
https://github.com/angular/angular.js/pull/16326/files#diff-85da78d5073ece90913c9e733cfcdef4R306

Yes, the test above not about `input` but I don't think that `input` tag is something special.
I have renamed `input` to `foo-input` to emphasize that name of the tag isn't matter

Reply:
That test is for a custom non-void element.
There should be a test that custom void element can be used as `<foo-input></foo-input>`.

Reply:
:up: Done